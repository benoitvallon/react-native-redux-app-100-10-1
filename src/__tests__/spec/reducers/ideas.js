'use strict';

import reducer from '../../../reducers/ideas';
import * as types from '../../../actions/actionTypes';

describe('ideas reducer', () => {
  it('should return the initial state', () => {
    expect(
      reducer(undefined, {})
    ).toEqual({
      ideas: []
    });
  });

  it('should return a wrong action', () => {
    expect(
      reducer(undefined, {
        type: types.WRONG,
        data: 'My first idea'
      })
    ).toMatchSnapshot();
  });

  it('should handle add and trim', () => {
    expect(
      reducer(undefined, {
        type: types.ADD,
        data: 'My first idea     '
      })
    ).toMatchSnapshot();

    expect(
      reducer({
        ideas: [
          { title: 'My first idea' }
        ]
      }, {
        type: types.ADD,
        data: 'My second idea \n\n\n '
      })
    ).toMatchSnapshot();
  });

  it('should handle remove', () => {
    expect(
      reducer({
        ideas: [
          { title: 'My first idea' },
          { title: 'My second idea' }
        ]
      }, {
        type: types.REMOVE,
        data: 1
      })
    ).toMatchSnapshot();
  });

  it('should handle save and trim', () => {
    expect(
      reducer({
        ideas: [
          { title: 'My second idea' }
        ]
      }, {
        type: types.SAVE,
        data: {
          index: 0,
          idea: 'My second idea edited \n\n\n '
        }
      })
    ).toMatchSnapshot();
  });

  it('should handle reset', () => {
    expect(
      reducer({
        ideas: [
          { title: 'My second idea edited' }
        ]
      }, {
        type: types.RESET
      })
    ).toMatchSnapshot();
  });
});
