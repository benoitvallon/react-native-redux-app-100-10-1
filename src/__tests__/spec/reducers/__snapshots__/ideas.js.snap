exports[`ideas reducer should handle add and trim 1`] = `
Object {
  "ideas": Array [
    Object {
      "title": "My first idea",
    },
  ],
}
`;

exports[`ideas reducer should handle add and trim 2`] = `
Object {
  "ideas": Array [
    Object {
      "title": "My first idea",
    },
    Object {
      "title": "My second idea",
    },
  ],
}
`;

exports[`ideas reducer should handle remove 1`] = `
Object {
  "ideas": Array [
    Object {
      "title": "My first idea",
    },
  ],
}
`;

exports[`ideas reducer should handle reset 1`] = `
Object {
  "ideas": Array [],
}
`;

exports[`ideas reducer should handle save and trim 1`] = `
Object {
  "ideas": Array [
    Object {
      "title": "My second idea edited",
    },
  ],
}
`;

exports[`ideas reducer should return a wrong action 1`] = `
Object {
  "ideas": Array [],
}
`;
